test/helpers/contract-helpers.ts

```diff
@@ -164,89 +165,165 @@ export const initializeContract = async () => {
 };
 
 export const liquidate = async (ownerAddress: string, operatorIds: number[], cluster: any) => {
-  const liquidatedCluster = await trackGas(DB.ssvNetwork.contract.liquidate(
-    ownerAddress,
-    operatorIds,
-    cluster
-  ));
+  const liquidatedCluster = await trackGas(DB.ssvNetwork.contract.liquidate(ownerAddress, operatorIds, cluster));
   return liquidatedCluster.eventsByName.ClusterLiquidated[0].args;
 };
 
 export const removeValidator = async (ownerId: number, pk: string, operatorIds: number[], cluster: any) => {
-  const removedValidator = await trackGas(DB.ssvNetwork.contract.connect(DB.owners[ownerId]).removeValidator(
-    pk,
-    operatorIds,
-    cluster
-  ));
+  const removedValidator = await trackGas(
+    DB.ssvNetwork.contract.connect(DB.owners[ownerId]).removeValidator(pk, operatorIds, cluster),
+  );
   return removedValidator.eventsByName.ValidatorRemoved[0].args;
 };
 
 export const withdraw = async (ownerId: number, operatorIds: number[], amount: string, cluster: any) => {
-  const withdrawnCluster = await trackGas(DB.ssvNetwork.contract.connect(DB.owners[ownerId]).withdraw(
-    operatorIds,
-    amount,
-    cluster));
+  const withdrawnCluster = await trackGas(
+    DB.ssvNetwork.contract.connect(DB.owners[ownerId]).withdraw(operatorIds, amount, cluster),
+  );
 
   return withdrawnCluster.eventsByName.ClusterWithdrawn[0].args;
 };
 
 export const reactivate = async (ownerId: number, operatorIds: number[], amount: string, cluster: any) => {
   await DB.ssvToken.connect(DB.owners[ownerId]).approve(DB.ssvNetwork.contract.address, amount);
-  const reactivatedCluster = await trackGas(DB.ssvNetwork.contract.connect(DB.owners[ownerId]).reactivate(
-    operatorIds,
-    amount,
-    cluster));
+  const reactivatedCluster = await trackGas(
+    DB.ssvNetwork.contract.connect(DB.owners[ownerId]).reactivate(operatorIds, amount, cluster),
+  );
   return reactivatedCluster.eventsByName.ClusterReactivated[0].args;
 };
 
-export const registerValidators = async (ownerId: number, numberOfValidators: number, amount: string, operatorIds: number[], gasGroups?: GasGroup[]) => {
+export const registerValidatorsWithKeystores = async (
+  ownerId: number,
+  amount: string,
+  keystores: string[],
+  passwords: string[],
+  operatorIds: number[],
+  operatorKeys: string[],
+  gasGroups?: GasGroup[],
+) => {
+  const validators: any = [];
+  let args: any;
+
+  if (keystores.length === passwords.length && operatorIds.length === operatorKeys.length) {
+    // keystores.length is same with the number of validators, because each validator has a corresponding keystore file.
+
+    const operators = operatorKeys.map((operatorKey, index) => ({
+      id: operatorIds[index],
+      operatorKey,
+    }));
+
+    const ssvKeys = new SSVKeys();
+
+    // Register validators to contract
+    for (let i = 0; i < keystores.length; i++) {
+      const { publicKey, privateKey } = await ssvKeys.extractKeys(keystores[i], passwords[i]);
```

Comment:
No iteration over keystores. publicKey, privateKey are retrieved from ValidatorsData JSON using the validator ID passed as parameter